internal/template/character/stam.go

```diff
@@ -42,18 +42,35 @@ func (c *Character) ActionStam(a action.Action, p map[string]int) float64 {
 }
 
 func (c *Character) Dash(p map[string]int) action.ActionInfo {
-	var f int = 20
-	switch c.CharBody {
-	case profile.BodyBoy, profile.BodyLoli:
-		f = 21
-	case profile.BodyGirl:
-		f = 20
-	case profile.BodyMale:
-		f = 19
-	case profile.BodyLady:
-		f = 22
+	length := c.DashLength()
+
+	// set the dash CD. If the dash was on CD when this dash executes, lockout dash
+	if c.Core.Player.DashCDExpirationFrame > c.Core.F {
+		c.Core.Player.DashLockout = true
+		c.Core.Player.DashCDExpirationFrame = c.Core.F + 1.5*60
+	} else {
+		c.Core.Player.DashLockout = false
+		c.Core.Player.DashCDExpirationFrame = c.Core.F + 0.8*60
+	}
+
+	c.Core.Log.NewEventBuildMsg(glog.LogCooldownEvent, c.Index, "dash cooldown triggered").
+		Write("lockout", c.Core.Player.DashLockout).
+		Write("expiry", c.Core.Player.DashCDExpirationFrame-c.Core.F).
+		Write("expiry_frame", c.Core.Player.DashCDExpirationFrame).
+		Write("dash_length", length)
```

Comment:
nit: Please move this to the branch with the actual dash cooldown above or make this into two separate logs with the 0.8*60 one informing the user that dash cd will be triggered if they dash before now + 0.8s and the 1.5*60 one actually saying "dash cooldown triggered". Having the same exact log message for both of these cases is confusing.